### PR TITLE
Add Delete state

### PR DESCRIPTION
**Pull request by chinzhiweiblank**
# Description

_Please explain the changes you've made_
Added Delete State to the DaprClient.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #74

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by youngbupark** [comment]
@chinzhiweiblank Thanks for this contribution. can you please address the comments that I left in https://github.com/dapr/python-sdk/pull/100 ? or please create the separate pr to resolve the comments in #100

File: dapr/clients/grpc/_helpers.py [hunk 1]
```diff
@@ -159,3 +160,35 @@ def intercept_unary_unary(
         # Call continuation
         response = continuation(new_call_details, request)
         return response
+
+
+def convert_string_to_duration(time: str) -> Duration:
```
Comment (youngbupark) [inline dapr/clients/grpc/_helpers.py:165]: We do not need anymore because we do not need retry options.

File: dapr/clients/grpc/_helpers.py [hunk 2]
```diff
@@ -10,8 +10,9 @@
 
 from google.protobuf.any_pb2 import Any as GrpcAny
 from google.protobuf.message import Message as GrpcMessage
+from google.protobuf.duration_pb2 import Duration
 from grpc import UnaryUnaryClientInterceptor, ClientCallDetails     # type: ignore
-
+import datetime
```
Comment (youngbupark) [inline dapr/clients/grpc/_helpers.py:15]: pls move import to line 7.

File: dapr/clients/grpc/_state.py [hunk 3]
```diff
@@ -1,35 +1,77 @@
...
 
 
+class RetryPattern(Enum):
+    """Represents the retry pattern for a Dapr State Api Call"""
+    unspecified = common_v1.StateRetryPolicy.RetryPattern.RETRY_UNSPECIFIED  # type: ignore
+    linear = common_v1.StateRetryPolicy.RetryPattern.RETRY_LINEAR  # type: ignore
+    exponential = common_v1.StateRetryPolicy.RetryPattern.RETRY_EXPONENTIAL  # type: ignore
+
+
+class RetryPolicy:
```
Comment (youngbupark) [inline dapr/clients/grpc/_state.py:28]: We do not need RetryPolicy which will be deprecated.

**Comment by chinzhiweiblank** [comment]
@youngbupark Actually I was wondering are there no integration tests for the python-sdk? Or is the fake dapr server sufficient for testing?

File: dapr/clients/grpc/client.py [hunk 4]
```diff
@@ -318,42 +341,110 @@ def save_state(
...
+                        ('header1', 'value1')
+                    ),
                 )
+
         Args:
             store_name (str): the state store name to save to
-            states (List[dict]): the key-value pairs to be saved
+            key (str): the key to be saved
+            value (bytes or str): the value to be saved
+            etag (str, optional): custom etag to save with
```
Comment (youngbupark) [inline dapr/clients/grpc/client.py:368]: etag is not a custom etag. etag must be given by backend statestore and used for save and delete. here it is given in the response payload.

**Comment by youngbupark** [comment]
> @youngbupark Actually I was wondering are there no integration tests for the python-sdk? Or is the fake dapr server sufficient for testing?

No integration test for python-sdk. fake server may not sufficient, but having integration tests is not easy in terms of maintenance. so that I didn't add integration test in the first place. If you can work on this, please create an issue.

**Comment by youngbupark** [comment]
@chinzhiweiblank Thanks for your contribution!

**Pull request merged**